### PR TITLE
Foundation: remove a number of non-portable POSIX codes

### PR DESCRIPTION
**Pull request by compnerd**
These are not available on Windows, do not use them.

### CODE REVIEW

**Comment by compnerd** [comment]
@colemancda not sure if this is what you had in mind

**Comment by millenomi** [comment]
LGTM, but @colemancda let me know if this is good by you.

File: Foundation/NSError.swift [hunk 1]
```diff
@@ -966,8 +966,10 @@ extension POSIXError {
     /// Bad address.
     public static var EFAULT: POSIXError.Code { return .EFAULT }
 
+    #if !os(Windows)
```
Comment (colemancda) [inline Foundation/NSError.swift]: I was doing
```
#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS) || os(Linux) || os(Android)
    /// Block device required.
    public static var ENOTBLK: POSIXError.Code { return .ENOTBLK }
    #endif
```

But yours is probably better xD

**Comment by colemancda** [comment]
@compnerd @millenomi Looks good!

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Pull request merged**